src/IonEventStream.ts

```diff
@@ -189,29 +191,39 @@ export class IonEventStream {
         return new ComparisonResult(ComparisonResultType.EQUAL);
     }
 
-    compareEquivs(expected: IonEventStream, comparisonReport: IonComparisonReport): ComparisonResult {
+    equivs(expected: IonEventStream, comparisonType: ComparisonType): boolean {
+        return this.compareEquivs(expected, comparisonType).result == ComparisonResultType.EQUAL;
+    }
+
+    compareEquivs(expected: IonEventStream , comparisonType: ComparisonType, comparisonReport?: IonComparisonReport): ComparisonResult {
         let actualIndex: number = 0;
         let expectedIndex: number = 0;
-        let comparisonType = comparisonReport.comparisonType;
+
         while (actualIndex < this.eventStream.length && expectedIndex < expected.eventStream.length) {
             let actualEvent = this.eventStream[actualIndex];
             let expectedEvent = expected.eventStream[expectedIndex];
 
             if(actualEvent.eventType == IonEventType.STREAM_END && expectedEvent.eventType == IonEventType.STREAM_END) {
                 break;
             } else if(actualEvent.eventType == IonEventType.STREAM_END || expectedEvent.eventType == IonEventType.STREAM_END) {
-                comparisonReport.writeComparisonReport(ComparisonResultType.ERROR,
-                    "Different number of comparison sets.", actualIndex, expectedIndex);
-                return new ComparisonResult(ComparisonResultType.ERROR);
+                    throw new Error("Different number of comparison sets.");
             } else if (!(actualEvent.ionType == IonTypes.LIST || actualEvent.ionType == IonTypes.SEXP)
                 || !(expectedEvent.ionType == IonTypes.LIST || expectedEvent.ionType == IonTypes.SEXP)) {
                 throw new Error("Comparison sets must be lists or s-expressions.");
-            } else if(this.isEmbedded as any ^ expected.isEmbedded as any) {
+            } else if((actualEvent.annotations[0] === 'embedded_documents') as any ^ (expectedEvent.annotations[0] === 'embedded_documents') as any) {
```

Comment:
> ```ts
> actualEvent.annotations[0] === 'embedded_documents') as any
> ```

Why do the boolean expressions need to be cast to `any`?

Since we check for 
```ts
someEvent.annotations[0] === 'embedded_documents'
```
several times in this code block, it might be worth making an `isEmbedded(event): boolean` helper function so the intent is obvious.

Reply:
For XOR operation in TS required both types to be any or number. it doesn't work with type boolean somehow.
And yes, will add isEmbedded as a helper function.